internal/clients/http/azure/types/azure_types.go

```diff
@@ -31,41 +29,14 @@ func init() {
 		panic(err)
 	}
 
-	// load supported types
-	supportedBuf, err := fsTypes.ReadFile("supported.yaml")
-	if err != nil {
-		panic(err)
-	}
-	err = typeInfo.RegisteredTypes.LoadSupported(supportedBuf)
-	if err != nil {
-		panic(err)
-	}
-
 	// load availability information
 	err = typeInfo.RegionalAvailability.Load(fsTypes, availabilityPath)
 	if err != nil {
 		panic(err)
 	}
 
-	// calculate etag
-	// concatenate all yamls into one 1 MiB buffer
-	dirEntries, err := fsTypes.ReadDir(availabilityPath)
-	if err != nil {
-		panic(err)
-	}
-	availBuf := bytes.NewBuffer(make([]byte, 0, 2^20))
-	for _, dirEntry := range dirEntries {
-		if dirEntry.IsDir() {
-			continue
-		}
-		file := filepath.Join(availabilityPath, dirEntry.Name())
-		buffer, errBuf := fsTypes.ReadFile(file)
-		if errBuf != nil {
-			panic(errBuf)
-		}
-		availBuf.Write(buffer)
-	}
-	etag, err = middleware.GenerateETagFromBuffer("azure-types", 30*time.Minute, typesBuf, supportedBuf, availBuf.Bytes())
+	availBuf := clients.ConcatBuffers(fsTypes, availabilityPath)
+	etag, err = middleware.GenerateETagFromBuffer("azure-types", 30*time.Minute, typesBuf, availBuf)
```

Comment:
I know this was there before, but why is there '30 minutes'? This could get less confusing using a constant.

Reply:
Absolutely valid point, it is the time all user browsers will consider the reply to cached. So when we roll out a new version of the application and any data changes within a list, the customers will see the change after 30 minutes in the worst case.

This was my wild guess, I was thinking when someone calls our support saying "my type is missing" so we would add it. Now that I think about it, I think we can do something like several hours. Let me extract this into a constant.